pkg/controller/atlasinstance/atlasinstance_controller.go

```diff
@@ -306,8 +306,7 @@ func (r *MongoDBAtlasInstanceReconciler) reconcileAtlasProject(cx context.Contex
 		if err != nil {
 			return
 		}
-		// Fetch the project name from newly created CR
-		atlasProject, err = r.getAtlasProject(cx, inst)
+		atlasProject = project
```

Comment:
What is the error that caused the operator pod to terminate? Function `r.getAtlasProject(cx, inst)` returns `nil`?

Reply:
Yes, it may return nil.

Reply:
If `r.getAtlasProject(cx, inst)` returns `nil` at line 294, the code will try to create the project at line 305.
If `r.getAtlasProject(cx, inst)` still returns `nil` at line 310, it means the creation fails, right? 
It this case, why creating the project at line 305 doesn't return error? I think `reconcile` should stop at line 305 as soon as the project creation fails.

Reply:
The issue I found was that after a creation call, a subsequent call of list/get may not return the object that was just created (seems there is a delay for the 2nd call ), and the list call in getAtlasProject() would not fail but simply return an empty result.